presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveBucketing.java

```diff
@@ -53,82 +49,84 @@
-        assertBucketEquals("varchar(15)", "");
-        assertBucketEquals("varchar(15)", "test string");
-        assertBucketEquals("varchar(15)", "\u5f3a\u5927\u7684Presto\u5f15\u64ce"); // 3-byte UTF-8 sequences (in Basic Plane, i.e. Plane 0)
-        assertBucketEquals("varchar(15)", "\uD843\uDFFC\uD843\uDFFD\uD843\uDFFE\uD843\uDFFF"); // 4 code points: 20FFC - 20FFF. 4-byte UTF-8 sequences in Supplementary Plane 2
-        assertBucketEquals("string", null);
-        assertBucketEquals("string", "");
-        assertBucketEquals("string", "test string");
-        assertBucketEquals("string", "\u5f3a\u5927\u7684Presto\u5f15\u64ce"); // 3-byte UTF-8 sequences (in Basic Plane, i.e. Plane 0)
-        assertBucketEquals("string", "\uD843\uDFFC\uD843\uDFFD\uD843\uDFFE\uD843\uDFFF"); // 4 code points: 20FFC - 20FFF. 4-byte UTF-8 sequences in Supplementary Plane 2
-        assertBucketEquals("date", null);
-        assertBucketEquals("date", new DateWritable(toIntExact(LocalDate.of(1970, 1, 1).toEpochDay())).get());
-        assertBucketEquals("date", new DateWritable(toIntExact(LocalDate.of(2015, 11, 19).toEpochDay())).get());
-        assertBucketEquals("date", new DateWritable(toIntExact(LocalDate.of(1950, 11, 19).toEpochDay())).get());
-        assertBucketEquals("timestamp", null);
-        assertBucketEquals("timestamp", new Timestamp(1000 * LocalDateTime.of(1970, 1, 1, 0, 0, 0, 0).toEpochSecond(ZoneOffset.UTC)));
-        assertBucketEquals("timestamp", new Timestamp(1000 * LocalDateTime.of(1969, 12, 31, 23, 59, 59, 999_000_000).toEpochSecond(ZoneOffset.UTC)));
-        assertBucketEquals("timestamp", new Timestamp(1000 * LocalDateTime.of(1950, 11, 19, 12, 34, 56, 789_000_000).toEpochSecond(ZoneOffset.UTC)));
-        assertBucketEquals("timestamp", new Timestamp(1000 * LocalDateTime.of(2015, 11, 19, 7, 6, 5, 432_000_000).toEpochSecond(ZoneOffset.UTC)));
-        assertBucketEquals("array<double>", null);
-        assertBucketEquals("array<boolean>", ImmutableList.of());
-        assertBucketEquals("array<smallint>", ImmutableList.of((short) 5, (short) 8, (short) 13));
-        assertBucketEquals("array<string>", ImmutableList.of("test1", "test2", "test3", "test4"));
-        assertBucketEquals("map<float,date>", null);
-        assertBucketEquals("map<double,timestamp>", ImmutableMap.of());
-        assertBucketEquals("map<string,bigint>", ImmutableMap.of("key", 123L, "key2", 123456789L, "key3", -123456L));
-        assertBucketEquals("array<array<bigint>>", ImmutableList.of(ImmutableList.of(10L, 20L), ImmutableList.of(-10L, -20L), asList((Object) null)));
-        assertBucketEquals("map<array<double>,map<int,timestamp>>", ImmutableMap.of(ImmutableList.of(12.3, 45.7), ImmutableMap.of(123, new Timestamp(1_234_567_890_000L))));
+        assertBucketEquals("boolean", null, 0);
+        assertBucketEquals("boolean", true, 1);
+        assertBucketEquals("boolean", false, 0);
+        assertBucketEquals("tinyint", null, 0);
```

Comment:
> `Fix Hive bucket calculation for timestamps with milliseconds`

Extract changes of `assertBucketEquals` to a prep commit.

Reply:
Thanks, I actually did have this separate, but I somehow lost it while rebasing. `git reflog` saves the day.